### ISSUE TITLE
There is also this package

### ISSUE THREAD

**Issue by wallymathieu** [spisemisu/SpiseMisu.SemanticVersioning#1]
https://github.com/wallymathieu/SemVer.FromAssembly

**Comment by gentauro** [comment]
Dude, why no ProjectScaffold + GitHub IO site? It's almost the same (well you don't handle NuGet pkgs and how do you handle pattern matching exhaustiveness?).

If this becomes a thing (fsproject) we should probably merge these projects together ...

> Note: Is your output markdown or JSON? I aimed to port Elm-packages bump and diff almost 1:1

**Comment by wallymathieu** [comment]
Well, I agree on Github IO site, it looks much more professional. I wrote the code while looking at the Elm-packages bump code as well. Some of the output is JSON in order to execute the tool from the tool in order to avoid problems with loading two assemblies with same name and version. 

We should definitely merge these two projects! It would be super sweet to have this for .net (with some nice documentation so that I can convince my coworkers this is a thing they should use).

**Comment by gentauro** [comment]
I gave a talk before releasing the library @ MF#K in Copenhagen:

- https://www.meetup.com/MoedegruppeFunktionelleKoebenhavnere/events/234664265/

And everybody were just like: "When do you release it", "I could so much use it for some of our projects." and "Do you have support for local NuGet repositories".

That's why I had to change my initial code before releasing :+1: 

> Note: I would love that somebody at Microsoft used Elm, just to understand why this is SO important for NuGet pkgs and libraries. My next step was to expand paket with the library if @forki wants ...

**Comment by wallymathieu** [comment]
Indeed! I agree, it would be sweet to have it in NuGet.exe and in paket.

**Comment by wallymathieu** [comment]
I could imagine that it would be useful for developers to be able to see the semver diff in the nuget update dialog in visual studio.

**Comment by gentauro** [comment]
Well I tried to arise awareness:

- https://twitter.com/genTauro42/status/794816223016054784

but is like @Microsoft don't really care :-1: 

But for me to catch on, like elm-package, it needs to be used with .NET packet manager, which is NuGet and paket (which also have push/pull packages logic).

> Note: You know you can specify Version Ranges right? https://docs.nuget.org/ndocs/create-packages/dependency-versions That way you will ensure that only Minor and Patch versions are automatically updated. Just like elm-package.

**Comment by forki** [comment]
We can talk about integration in paket, but in my experience these
heuristics are more wrong than correct. And thats not implementation fault,
but the simple fact that all changes influence if something is actually
breaking - surface area is only one small part of it.

Am 18.12.2016 10:02 schrieb "Ramón Soto Mathiesen" <notifications@github.com

**Comment by gentauro** [comment]
And that's why Semantic Versioning is actually a bad name for this. It should be Syntactic Versioning ... That is the problem that is solved with this library (and all other SemVer out there) :+1:

**Comment by forki** [comment]
Yes. Absolutely. This would make it actually a really useful tool for
maintainers.

Am 18.12.2016 10:07 schrieb "Ramón Soto Mathiesen" <notifications@github.com

**Comment by gentauro** [comment]
Another question @forki, as we both have tried to implement SemVer, I think it would give sense to merge them into one project. How do we request to get it as a @fsprojects? Afterwards, we will try to integrate to paket :+1:

**Comment by forki** [comment]
Just let me know via twitter dm or any other private channel. I have all
the power needed ;-)

Am 18.12.2016 10:11 schrieb "Ramón Soto Mathiesen" <notifications@github.com

**Comment by gentauro** [comment]
> Just let me know via twitter dm or any other private channel. I have all the power needed ;-)

 wallymathieu, lets do it :+1:

**Comment by wallymathieu** [comment]
Indeed!

**Comment by gentauro** [comment]
@wallymathieu we can't send you DM on Twitter :-1: 

Here is what we have said:

---
Steffen Forkmann:
You need to make me admin on the project. And I need to know the github handles of the admins
6m

Ramón Soto Mathiesen:
Hi Steffen
6m

Ramón Soto Mathiesen:
I was thinking that we both have different namespaces so it might be easier to create a new project under fsprojects and just merge code?

Ramón Soto Mathiesen:
You have tried this before, what is the best approach?
5m

Steffen Forkmann:
That will lose all github issue history

Steffen Forkmann:
And watchers
4m 11 seconds ago

Ramón Soto Mathiesen:
I have 3 stars and Oskar has 5. I think we can live with it :)
3m

Steffen Forkmann:
Ok
2m

Ramón Soto Mathiesen:
btw I would add him to these messages, but I'm not allowed to send him private messages ...
2m

Steffen Forkmann:
Ok then just copy these messages and open discussion issue

Steffen Forkmann:
When things are decided we meet in private channel again
34s

Ramón Soto Mathiesen:
I guess if you can create: fsprojects/SemVer where you stay as admin and skar and I are contributors, that would be fine

Ramón Soto Mathiesen:
Roger that
---

Do you agree?

**Comment by wallymathieu** [comment]
Sure, works for me!

**Comment by wallymathieu** [comment]
I've added you ( https://twitter.com/gentauro42 ? ) to twitter https://twitter.com/ozzymcduff

**Comment by gentauro** [comment]
Nice.

Btw when the project is created. Lets have an open dialogue on how we are going to do stuff (just with issues on GitHub).

I think it will be a good learning experience for other to watch/participate. Do you agree?

**Comment by wallymathieu** [comment]
Sure 👍

**Comment by forki** [comment]
so what's the name? "SemVer"

**Comment by wallymathieu** [comment]
I think that the thing to take out of https://github.com/wallymathieu/SemVer.FromAssembly is the usage of a separate command line execution of a process with assembly load. I.e. a lot of focus on getting it to work with albacore, cake, fake for existing assemblies. The code for diffs that you have done looks more in geared towards .net differences: properties, enum values, et.c. so is probably more structured and appropriate for .net than the diff code I've written (I used the Elm bump + f# surface area test code as a basis).

**Comment by wallymathieu** [comment]
Looks like there is already a project with that name on nuget: https://www.nuget.org/packages/semver/

**Comment by gentauro** [comment]
> so what's the name? "SemVer"

FsProjects.SemVer? That way we avoid mistaken people when retrieving the library

**Comment by forki** [comment]
FsProjects shouldn't be part of the project name.

**Comment by gentauro** [comment]
@wallymathieu I like your approach of using a standalone executable, but I think we still need a lib under the hood (the standalone should just be a wrapper around the library).

**Comment by wallymathieu** [comment]
@gentauro the exe is mostly in order to avoid the problem of loading two assemblies with the same name and version from different locations.

**Comment by gentauro** [comment]
> Looks like there is already a project with that name on nuget: https://www.nuget.org/packages/semver/

https://www.nuget.org/packages/semanticversioning/ (is taken as well)

Should we just go with **SyntacticVersioning**, which is really what is it? We can still argue about the name and why we have chosen it.

https://www.nuget.org/packages/syntacticversioning/ (it's available)

**Comment by gentauro** [comment]
> @gentauro the exe is mostly in order to avoid the problem of loading two assemblies with the same name and version from different locations.

I think I will need an example to understand this :)

**Comment by wallymathieu** [comment]
In your current nuget package you have a dll with name ```Something``` with assembly version ```1.0.0``` in order to figure out the next version of that dll while building it you load the nuget-dll via Assembly.Load and then the currently compiled ```Something``` with assembly version ```1.0.0``` in order to figure out the differences. What happens? The way to do it in .net would be to load one of the assemblies in one AppDomain and the other in another AppDomain (from what I understand). However, there have been a lot of weirdness in mono related to AppDomain. I think I've read that AppDomain is currently not going to be supported in .net core.

**Comment by gentauro** [comment]
"There are only two hard things in Computer Science: cache invalidation and **naming things**." -- Phil Karlto

**Comment by wallymathieu** [comment]
I had the problem that the mono runtime did not want to load the same assembly twice from different locations
https://msdn.microsoft.com/en-us/library/dd153782.aspx#avoid_loading_multiple_versions
however, looks like it works differently in the windows .net runtime (based on what they say in the article). Still, question is if this behaviour will be consistent (since it's related to something few devs use)?

**Comment by gentauro** [comment]
I'm on Linux and have developed everything (and done my test) on that platform. I don't even recall how Windows work anymore ;)

So I don't think it's an issue, some of my test load the same assembly (file location downloaded from NuGet) vs. the NuGet package directly (I'm using Assembly.LoadFile) without any issues.

> Should we just go with SyntacticVersioning

Are you keen with that name?

**Comment by wallymathieu** [comment]
SyntacticVersioning works fine

**Comment by wallymathieu** [comment]
When I try to do that on mac os x, mono version 4.6.2 I get a patch difference when it should return major change. What version of mono are you using? Have they fixed it in 4.8 ?

**Comment by wallymathieu** [comment]
```
To avoid the duplication of the NuGet packages, can I suggest that you follow the deprecation process outlined here:

https://chocolatey.org/docs/how-to-deprecate-a-chocolatey-package

Yes, that is the process followed for Chocolatey Packages, but the same process applies equally to NuGet Packages.
```
from https://github.com/cake-build/website/pull/307#issuecomment-287583415
